Source/Core/Core/AchievementManager.cpp

```diff
@@ -370,32 +353,20 @@ bool AchievementManager::CanPause()
 
 void AchievementManager::DoIdle()
 {
+  rc_client_set_allow_background_memory_reads(m_client, 1);
   std::thread([this] {
     while (true)
     {
-      Common::SleepCurrentThread(1000);
+      Common::SleepCurrentThread(100);
+      std::lock_guard lg{m_lock};
+      Core::System* system = m_system.load(std::memory_order_acquire);
+      if ((!system || Core::GetState(*system) != Core::State::Paused) ||
+          (!m_background_execution_allowed) || (!m_client || (!IsGameLoaded() && !m_dll_found)))
       {
-        std::lock_guard lg{m_lock};
-        Core::System* system = m_system.load(std::memory_order_acquire);
-        if (!system || Core::GetState(*system) != Core::State::Paused)
-          return;
-        if (!m_background_execution_allowed)
-          return;
-        if (!m_client || !IsGameLoaded())
-          return;
+        rc_client_set_allow_background_memory_reads(m_client, 0);
+        return;
       }
-      // rc_client_idle peeks at memory to recalculate rich presence and therefore
-      // needs to be on host or CPU thread to access memory.
-      Core::QueueHostJob([this](Core::System& system) {
-        std::lock_guard lg{m_lock};
-        if (Core::GetState(system) != Core::State::Paused)
-          return;
-        if (!m_background_execution_allowed)
-          return;
-        if (!m_client || !IsGameLoaded())
-          return;
-        rc_client_idle(m_client);
-      });
+      rc_client_idle(m_client);
```

Comment:
`rc_client_idle` can call MemoryPeeker, right? If so, calls to it need to be on the CPU thread or host thread.